docs/design.md

```diff
@@ -104,3 +104,47 @@ several ways according to policy:
 
 Then like the usual LibOS pattern it will call the container's entry point as though the container
 was a process unto itself (rather than where it now sits like a library in the LibOS PAL process).
+
+## [Future Work] Local Attestation for Filesystem Decryption Key Provisioning
+
+![EnclaveCC_LA.png](images/EnclaveCC_LA.png)
+
+A local attestation (LA) feature is needed for authentication/authorization and secret provisioning
+between an Agent-enclave instance and a set of App-enclave instances resided in a same Pod. The
+agent holds a secret for filesystem decryption and the app wants to authenticate to the agent in
+order for the agent to release the filesystem decryption key (FDK). The app will prove its identity
+with a report and the agent will check that against a policy which requires a matching `MRSIGNER` or
+`MRENCLAVE`, possibly including an additional value from KSS. This design is a guide for enclave-CC
+developers to implement this feature.
+
+In this design, we leverages the `AEAD` algorithm for FDK protection over `ttrpc` channel. It is a
+recommended cryptographic algorithm in IETF, see [rfc5116](https://datatracker.ietf.org/doc/html/rfc5116).
+Regarding the ephemeral Diffie-Hellman key exchange, we chooses the Curve25519, which is recommended
+by IETF [rfc8031](https://www.rfc-editor.org/rfc/rfc8031). The design doesn't include the agent
+authentication in enclave-CC security context, however, the app can easily add app side user defined
+policy to enhance the security configuration.
+
+In most scenarios, it is one-shot transactions used for single filesystem decryption key (FDK)
+provisioning. However, this LA protocol can easily be extended to provision multiple keys for more
+general purposes.
+
+To achieve this, it is necessary to add a key id to the second request message sent along G_a and
+app's report, with the ID being used to indicate which key is being requested. But whether the key
```

Comment:
```suggestion
app's report, with the key id being used to indicate which key is being requested. But whether the key
```

Reply:
it looks `key id` is commonly used in this paragraph

Reply:
Agree, thanks.